config/validator.go

```diff
@@ -142,3 +142,40 @@ func RequestHeaderContentTypeValidator(req http.Request, requiredContentType str
 	}
 	return 0
 }
+
+
+// DomainsEntryLogicalValidator checks the correctness of "Domains" part of configuration file
+func (c *YamlConfig) DomainsEntryLogicalValidator() (valid bool, validationErrors map[string][]error) {
+	errList := make([]error, 0)
+	var domains []string
+	for _, regionConf := range c.Regions {
+		for _, domain := range regionConf.Domains {
+			for _, alreadyDefinedDomain := range domains {
```

Comment:
You may move body of that for bloc to separate function for clarity (one indentation less)
eg func isAlreadyDefined(current string, oldones []strings)